### PR TITLE
Python: Update import HnswAlgorithmConfiguration

### PR DESCRIPTION
**Pull request by seb-jul**
### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

File: python/semantic_kernel/connectors/memory/azure_cognitive_search/azure_cognitive_search_memory_store.py [hunk 1]
```diff
@@ -8,7 +8,7 @@
 from azure.core.exceptions import ResourceNotFoundError
 from azure.search.documents.indexes.aio import SearchIndexClient
 from azure.search.documents.indexes.models import (
-    HnswVectorSearchAlgorithmConfiguration,
+    HnswAlgorithmConfiguration,
```
Comment (moonbox3) [inline python/semantic_kernel/connectors/memory/azure_cognitive_search/azure_cognitive_search_memory_store.py:11]: The [unit tests](https://github.com/microsoft/semantic-kernel/actions/runs/8021982266/job/21921710484?pr=5130#step:5:32) are failing for Python 3.8.

**Review by moonbox3 — CHANGES_REQUESTED** [review]

**Comment by moonbox3** [comment]
Does this require a package update to `11.6.0b1`?

**Comment by seb-jul** [comment]
Yes https://github.com/Azure/azure-sdk-for-python/blob/azure-search-documents_11.6.0b1/sdk/search/azure-search-documents/CHANGELOG.md

**Comment by seb-jul** [comment]
`11.6.0b1` is a prerelease but I see it's allowed for azure-search-documents

**Comment by seb-jul** [comment]
@microsoft-github-policy-service agree

**Comment by moonbox3** [comment]
Looks like the current code still doesn't play well with Python 3.8.

**Review by moonbox3 — APPROVED** [review]

**Review by eavanvalkenburg — APPROVED** [review]
thanks for keeping this one up to date!

**Comment by moonbox3** [comment]
Hi @seb-jul, we had a lot of changes go into master in the last few days. Locally, could you please pull the latest down, merge, and update the poetry lock file as stated [here](https://github.com/microsoft/semantic-kernel/actions/runs/8065936467/job/22035354310?pr=5130#step:6:14)? Thank you!

**Comment by seb-jul** [comment]
Hello @moonbox3 just pushed updated poetry lock file

**Comment by eavanvalkenburg** [comment]
@seb-jul can you have a look at the failing tests?

**Comment by seb-jul** [comment]
https://github.com/microsoft/semantic-kernel/pull/5223 just created another PR sorry

**Pull request closed without merging**